main.cpp

```diff
@@ -649,6 +657,59 @@ void part4()
     std::cout << "---------------------\n" << std::endl;
 }
 
+void myFloatFreeFunct(float& f)
+{
+    f += 7.0f;
+}
+
+void myDoubleFreeFunct(double& d)
+{
+    d += 6.0;
+}
+
+void myIntFreeFunct(int& i)
+{
+    i += 5;
+}
+
+
+void part6()
+{
+    FloatType ft3(3.0f);
+    DoubleType dt3(4.0);
+    IntType it3(5);
+    
+    std::cout << "Calling FloatType::apply() using a lambda (adds 7.0f) and FloatType as return type:" << std::endl;
+    std::cout << "ft3 before: " << ft3 << std::endl;
+    ft3.apply( [&](float&) ->FloatType& {return ft3 += 7.0f;});
```

Comment:
2 things:
- Don't capture more than you need to. this lambda is capturing `ft3`, `dt3` and `it3` and it only uses `ft3`
- While we know the `operator+=` returns `*this`, the way you have written this `return` statement is confusing as to what's being returned.  Always aim for readability over conciseness.
write:
```
{
    ft3 += 7.0f;
    return ft3;
}
```
- [ ] revise